### PR TITLE
fix(stepfunctions-tasks): instance type cannot be provided to SageMakerCreateTransformJob as input path

### PR DESCRIPTION
**Pull request by syeehyn**
as referred in https://github.com/aws/aws-cdk/issues/11605 in SageMakerCreateTransformJob has the same kind of issue.

similar solution can be found at https://github.com/aws/aws-cdk/pull/11749

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/create-transform-job.ts [hunk 1]
```diff
@@ -227,7 +227,8 @@ export class SageMakerCreateTransformJob extends sfn.TaskStateBase {
     return {
       TransformResources: {
         InstanceCount: resources.instanceCount,
-        InstanceType: 'ml.' + resources.instanceType,
+        InstanceType: sfn.JsonPath.isEncodedJsonPath(resources.instanceType.toString())
+        ? resources.instanceType.toString() : `ml.${resources.instanceType}`,
```
Comment (BenChaimberg) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/create-transform-job.ts:231]:
```suggestion
          ? resources.instanceType.toString() : `ml.${resources.instanceType}`,
```

**Review by BenChaimberg — CHANGES_REQUESTED** [review]
Please add test(s) to ensure both ways of providing an instance type produce a valid output

**Comment by syeehyn** [comment]
> Please add test(s) to ensure both ways of providing an instance type produce a valid output

Added test for the jsonString path. w.r.t this PR https://github.com/aws/aws-cdk/pull/14064

File: packages/@aws-cdk/aws-stepfunctions-tasks/test/sagemaker/create-training-job.test.ts [hunk 2]
```diff
@@ -401,3 +401,4 @@ test('Cannot create a SageMaker train task with both algorithm name and image na
   }))
     .toThrowError(/Must define either an algorithm name or training image URI in the algorithm specification/);
 });
+
```
Comment (BenChaimberg) [inline packages/@aws-cdk/aws-stepfunctions-tasks/test/sagemaker/create-training-job.test.ts:404]:
```suggestion
```

**Review by BenChaimberg — APPROVED** [review]

**Pull request merged**